Create 2021-07-21-handlingmemorymessages.md (#345)
micro-ROS/micro-ROS.github.io: _posts/2021-07-21-handlingmemorymessages.md

---

> title: Handling memory messages Tutorial for micro-ROS

Reviewer: ```suggestion
title: Handling Messages Memory Tutorial for micro-ROS
```

author: Maria Merlan
---
Having total awareness of what ROS 2 messages and types are being used for, is essential in order to handle memory correctly in an embedded C99 environment. 

The following Tutorial “Handling messages memory in micro-ROS '' at [icro-ROS web page](https://micro.ros.org/docs/tutorials/advanced/handling_type_memory/) explains in detail how to handle messages and types memory in micro-ROS. 

> The two approaches of micro-ROS memory handling are present in this tutorial.
> 
> micro-ROS Foxy: Memory handling using traditional allocation approach.
> 
> micro-ROS Galactic: Memory handling using type support introspection.

Reviewer: ```suggestion
The two approaches of micro-ROS memory handling are present in this tutorial:
* micro-ROS Foxy: Memory handling using traditional allocation approach.
* micro-ROS Galactic: Memory handling using type support introspection.
```

> Type Support introspection in C in micro-ROS Galactic distribution enables the new type handling API. This package is able to auto-assign memory to a certain message structure using default dynamic memory allocators. 

Reviewer: ```suggestion
Type support introspection in C in the micro-ROS Galactic distribution is enabled by the new type handling API. This package is able to auto-assign memory to a certain message structure using default dynamic memory allocators. 
```
